experiment/simulation/js/animation-DFF.js

```diff
@@ -0,0 +1,404 @@
+];
+// First 2 dots emerge from Data
+// Third dot is from Clock
+
+
+
+// decide help to decide the speed
+let decide = false;
+// circuitStarted is initialised to 0 which depicts that demo hasn't started whereas circuitStarted 1 depicts that the demo has started.
+let circuitStarted = false;
+
+
+// function to take care of width
+function demoWidth() {
+    if (width < 1024) {
+        circuitBoard.style.height = "600px";
+    } else {
+        circuitBoard.style.height = `${windowHeight - circuitBoardTop - 20}px`;
+    }
+    sidePanels[0].style.height = circuitBoard.style.height;
+}
+
+// function to initialise the input text i.e. either 0/1 that gets displayed after user click on them
+function textIOInit() {
+    for( const text of textInput){
+        text.textContent = 2;
+    }
+}
+
+
+// function to mark the output coordinates
+function outputCoordinates() {
+    setCoordinates(896,424,textOutput[0]);
+    svg.append(textOutput[0]);
+    setCoordinates(896,604,textOutput[1]);
+    svg.append(textOutput[1]);
+}
+
+// function to mark the input dots
+function inputDots() {
+    for(const dot of dots){
+        fillInputDots(dot,20,550,15,"#FF0000");
+        svg.append(dot);
+    }
+}
+
+// function to disappear the input dots
+function inputDotDisappear() {
+    for(const dot of dots){
+        objectDisappear(dot);
+    }
+}
+
+// function to appear the input dots
+function inputDotVisible() {
+    for(const dot of dots){
+        objectAppear(dot);
+    }
+}
+// function to disappear the output text
+function outputDisappear() {
+    for(const text of textOutput){
+        objectDisappear(text);
+    }
+}
+// function to appear the output text
+function outputVisible() {
+    for(const text of textOutput){
+        objectAppear(text);
+    }
+}
+// function to diappear the input text
+function inputTextDisappear() {
+    for(const text of textInput){
+        objectDisappear(text);
+    }
+}
+
+function clearObservation() {
+    observ.innerHTML = EMPTY;
+}
+function allDisappear() {
+    inputDotDisappear();
+    outputDisappear();
+    inputTextDisappear();
+    for(const object of objects){
+        fillColor(object,"#008000");
+    }
+}
+
+function appendData() {
+    if (textInput[0].textContent !== "0" && timeline.progress() === 0) {
+        changeto0(16,124,0,0);
+        observ.innerHTML = "Data bit is equal to 0";
+    }
+    else if (textInput[0].textContent !== "1" && timeline.progress() === 0) {
+        changeto1(16,124,0,0);
+        observ.innerHTML = "Data bit is equal to 1";
+    }
+    setter(textInput[0].textContent,dots[0]);
+    setter(textInput[0].textContent,dots[1]);
+}
+function appendClock() {
+    if (textInput[1].textContent !== "0" && timeline.progress() === 0) {
+        changeto0(16,514,1,1);
+    }
+    else if (textInput[1].textContent !== "1" && timeline.progress() === 0) {
+        changeto1(16,514,1,1);
+        observ.innerHTML = "Clock is turned ON";
+    }
```

Comment:
Please see if a default else block is needed

Reply:
Will have to leave the else block empty, it is not an error, we just can't change the values when timeline.progress isn't 0.

Reply:
Added the else block, it displays a message in the observation table